### PR TITLE
PF Session 1 Assignment 1 13/7/17

### PR DESCRIPTION
**Pull request by meta-shreya-bordia**
Sir,
I have duly filled in the PF1 Session 1 Assignment. Kindly review it.
Thank you.
Shreya Bordia

### CODE REVIEW

File: PF1/assignment1/BinaryToOctal.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
+package assignment1;
+
+//convert binary number to octal format
+public class BinaryToOctal {
+	int convertBinaryToOctal(int n){
+		int no= n;
+		while(no > 10) {
```
Comment (surendra2207) [inline PF1/assignment1/BinaryToOctal.java]: Include line gap when you start any new block of code for code readability.

File: PF1/assignment1/BinaryToOctal.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
+package assignment1;
+
+//convert binary number to octal format
+public class BinaryToOctal {
+	int convertBinaryToOctal(int n){
+		int no= n;
```
Comment (surendra2207) [inline PF1/assignment1/BinaryToOctal.java]: Include line gap when you start any new block of code for code readability. Make similar changes in all assignment in future.

File: PF1/assignment1/BinaryToOctal.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+	int convertToDecimal(int binary_no){
+		int decimal_no = 0;
+		
+		int power=0;
+		while(binary_no>0){
+			decimal_no += (binary_no%10)*((int)Math.pow(2, power++));
+			binary_no = binary_no/10;
+		}
+		return decimal_no;
+
```
Comment (surendra2207) [inline PF1/assignment1/BinaryToOctal.java]: Do not put extra line gaps unnecessarily. Please correct on all places as well.

File: PF1/assignment2/RemoveDuplicate.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
+package assignment2;
+
+// find unique elements of an array
+public class RemoveDuplicate {
+	
+	/* Replace each repeating element with
+	 * maximum possible integer size 
+	 * */
+	public int[] removeDuplicate(int[] n) {
```
Comment (surendra2207) [inline PF1/assignment2/RemoveDuplicate.java]: Needs to include null handling in this menthod.

File: PF1/assignment2/RemoveDuplicate.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+		int[] tmp= new int[n.length];			
+		int k=0;
+		final int MAX = Integer.MAX_VALUE;
+		
+		for(int i=0; i< n.length; i++) {
+			if(n[i] != MAX) {
+				for(int j=i+1; j<n.length; j++) {
+					if(n[j] != MAX){
```
Comment (surendra2207) [inline PF1/assignment2/RemoveDuplicate.java:23]: Use Logic (&&) operator to combine both if blocks if(n[j] != MAX && n[i] == n[j])

File: PF1/assignment2/RemoveDuplicateTest.java [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+		int[] expected = {};
+		
+		//check for empty arrays
+		Assert.assertArrayEquals(expected, duplicate_element.removeDuplicate(input));
+		
+	}
+	
+	//testing for a random number of duplicates
+	@Test
+	public void testForCase(){
```
Comment (surendra2207) [inline PF1/assignment2/RemoveDuplicateTest.java]: method name should be meaningful.

File: PF1/assignment3/LongestSequence.java [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+	public int[] longestSequence(int[] input){
+		if(input.length <0)
+			throw new ArrayIndexOutOfBoundsException();
+		
+		if(input.length ==1) 
+			return input;
+		
+		int[] tmp= new int[input.length];
+		int k=0;
+
```
Comment (surendra2207) [inline PF1/assignment3/LongestSequence.java]: Variable declaration should be at start of each block of code.

File: PF1/assignment3/LongestSequence.java [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+				/* break the sequence */
+				end_marker = i;
+				
+				int len_seq = end_marker - start_marker+1;
+				if(len_seq >= max){
+					max = len_seq;
+					
+					/*Sub Sequence*/
+					k=0;
+					for(int x = start_marker; x< end_marker+1; x++){
```
Comment (surendra2207) [inline PF1/assignment3/LongestSequence.java:37]: For loop at 34 line and 52 line need to be moved to single menthod to reusability.

File: PF1/assignment3/LongestSequenceTest.java [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+	@Test
+	public void testSameElements(){
+		int[] input = {5,5,5,5,5,5,5,5,5,5,5,5,5,5};
+		int[] expected= {5};
+		int[] actual = find_sequence.longestSequence(input);
+		
+		Assert.assertArrayEquals(expected, actual);
+	}
+	
+	@Test
```
Comment (surendra2207) [inline PF1/assignment3/LongestSequenceTest.java:44]: Provide comment over each test method.

File: PF1/assignment3/LongestSequenceTest.java [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+	public void testOneElement(){
+		int[] input = {4};
+		int[] expected= {4};
+		int[] actual = find_sequence.longestSequence(input);
+		
+		Assert.assertArrayEquals(expected, actual);
+	}
+	
+	@Test
```
Comment (surendra2207) [inline PF1/assignment3/LongestSequenceTest.java:53]: Provide comment over each test method.

File: PF1/assignment3/LongestSequenceTest.java [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+	public void testForBlank(){
+		int[] input = {};
+		int[] expected= {};
+		int[] actual = find_sequence.longestSequence(input);
+		
+		Assert.assertArrayEquals(expected, actual);
+	}
+	
+	@Test(expected = NullPointerException.class)
```
Comment (surendra2207) [inline PF1/assignment3/LongestSequenceTest.java]: Provide comment over each test method.

File: PF1/assignment5/CheckArraySorting.java [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+		/* => is descending sorted*/
+		if(i == a.length- 1){
+			return 2;	 
+		}
+			
+		return 0;
+		}
+}			
+		
+/**		Another algorithm
```
Comment (surendra2207) [inline PF1/assignment5/CheckArraySorting.java:42]: Remove commented code

File: PF1/assignment6/MergeArrays.java [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+		
+		if(a[0]<=a[1]){
+			while(i<a_size & j<b_size){
+				if(a[i] <= b[j]){
+					c[k++] = a[i++];
+				}else{
+					c[k++] = b[j++];
+				}
+			}
+
```
Comment (surendra2207) [inline PF1/assignment6/MergeArrays.java:29]: Need to move code from line  24-30 out of if else so no need to write same code from line 39-44.

File: PF1/assignment2/RemoveDuplicate.java (hunk 5 above)
Comment (meta-shreya-bordia) [reply PF1/assignment2/RemoveDuplicate.java:23]: The changes requested have been committed.

File: PF1/assignment2/RemoveDuplicateTest.java (hunk 6 above)
Comment (meta-shreya-bordia) [reply PF1/assignment2/RemoveDuplicateTest.java]: I have changed the case names to proper meaningful ones.

File: PF1/assignment3/LongestSequenceTest.java (hunk 9 above)
Comment (meta-shreya-bordia) [reply PF1/assignment3/LongestSequenceTest.java:44]: The changes requested have been made.

File: PF1/assignment5/CheckArraySorting.java (hunk 12 above)
Comment (meta-shreya-bordia) [reply PF1/assignment5/CheckArraySorting.java:42]: This code included different approach to same problem solving.
All extra comments have been removed.

**Review by surendra2207 — APPROVED** [review]

**Pull request merged**